pages/pipelines/buildkite_secrets.md

```diff
@@ -0,0 +1,85 @@
+    * If you attempt to use any other characters for the key, or you begin your key with `buildkite` or `bk` (regardless of case), your secret will not be created when selecting **Create Secret**.
+
+1. Enter the **Value** for the secret. This value can be any number of valid UTF-8 characters up to a maximum of 8 kilobytes. Be aware that once the secret is created, this value will no longer be visible through the Buildkite interface and will be redacted when output in build logs.
+1. Select **Create Secret** to create your new secret, which can now be referenced and used in build steps and jobs, respectively.
+
+## Use a Buildkite secret in a job
+
+### From a build script or hook
+
+Once you've [created a secret](#create-a-secret), the [`buildkite-agent secret get` command](/docs/agent/v3/cli-secret) can be used within the Buildkite Agent to print the secret's value to standard out (stdout). You can use this command within standard bash-like commands to redirect the secret's output into an environment variable, a file, or your own tool that uses the Buildkite secret's value directly, for example:
+
+- Setting a Buildkite secret with the key `secret_name` into an environment variable called `SECRET_VAR`:
+
+    `SECRET_VAR=$(buildkite-agent secret get secret_name)`
+
+- Redirecting the value of a Buildkite secret with the key `secret_name` into a file called `secret.txt`:
+
+    `buildkite-agent secret get secret_name > secret.txt`
+
+- Passing the output of your Buildkite secret (via the `buildkite-agent secret get` command) to your own tool named `cli-tool` that accepts a secret via its `-token` option:
+
+    `cli-tool —token $(buildkite-agent secret get secret_name)`
+
+Here’s a simple example of how one of these commands might appear in a Buildkite Pipeline step:
+
+```yaml
+steps:
+  - agents: { queue: "hosted" }
+    command:
+      - buildkite-agent secret get secret_name > secret.txt
+```
+{: codeblock-file="pipeline.yml"}
+
+## Redaction
+
+If any Pipeline, script or your own tool (accidentally) prints out the value of any Buildkite secret to standard out, this value is automatically redacted from the build logs.
+
+If for any reason you detect a secret value that isn't redacted, please rotate your secrets and contact security@buildkite.com.
+
+## Security
+
+Buildkite secrets are designed, with the following controls in place:
+
+- Secrets are encrypted in transit using TLS.
+- Secrets are always stored encrypted at rest.
+- All access to the secrets are logged.
+- Employee access to secrets is strictly limited and audited.
+
+> 📘 Buildkite secrets is not a zero-knowledge system
+> Buildkite owns, stores and manages the keys used for encrypting the secrets stored in the service at rest and in transit. Buildkite recommend customers implement additional controls to manage the lifecycle of secrets stored within Buildkite secrets, in addition to any monitoring capability you may require in line with your risk appetite.
```

Comment:
we give customers the necessary information here, but i think we should draw the conclusion as well. absolute candor, and all that

```suggestion
> Buildkite owns, stores and manages the keys used for encrypting the secrets stored in the service at rest and in transit. **This means that were Buildkite to be compromised, your secrets could be compromised as well.** Buildkite recommend customers implement additional controls to manage the lifecycle of secrets stored within Buildkite secrets, in addition to any monitoring capability you may require in line with your risk appetite.
```